deploy/tekton/tasks/execute_sast_ai_workflow.yaml

```diff
@@ -661,17 +696,104 @@ spec:
           exit 1
         fi
 
-    # STEP 8: Cleanup (Always runs)
+    # STEP 8: Upload SARIF to GCS Bucket
+    - name: upload-sarif-to-gcs
+      image: google/cloud-sdk:slim
+      env:
+        - name: GCS_BUCKET_NAME
+          value: "$(params.GCS_BUCKET_NAME)"
+        - name: GCS_BUCKET_UPLOAD_ENABLED
+          value: "$(params.GCS_BUCKET_UPLOAD_ENABLED)"
+        - name: GOOGLE_APPLICATION_CREDENTIALS
+          value: "$(workspaces.gcs-sa-json-ws.path)/$(params.GCS_SA_FILE_NAME)"
+        - name: PROJECT_NAME
+          value: "$(params.PROJECT_NAME)"
+        - name: PROJECT_VERSION
+          value: "$(params.PROJECT_VERSION)"
+      volumeMounts:
+        - name: shared-data
+          mountPath: /shared-data
+        - name: gcs-scripts
+          mountPath: /scripts
+      script: |
+        #!/bin/bash
+        set -e
+        echo "=== STEP 8: UPLOAD SARIF TO GCS BUCKET ==="
+        
+        # Check if GCS upload is enabled
+        if [ "$GCS_BUCKET_UPLOAD_ENABLED" != "true" ]; then
+          echo "GCS bucket upload disabled - skipping"
+          exit 0
+        fi
+        
+        # Check if we have required parameters
+        if [ -z "$GCS_BUCKET_NAME" ]; then
+          echo "Skipping GCS upload - no bucket name provided"
+          exit 0
+        fi
+        
+        # Check service account
+        if [ ! -f "$GOOGLE_APPLICATION_CREDENTIALS" ]; then
+          echo "Skipping GCS upload - service account not available"
+          exit 0
+        fi
+        
+        # Look for the SARIF file in output directory
+        SARIF_FILE=$(find /shared-data/output -name "*.sarif" -type f 2>/dev/null | head -1)
+        
+        if [ -z "$SARIF_FILE" ]; then
+          echo "No SARIF file found in output directory - skipping GCS upload"
+          echo "Available files in output directory:"
+          ls -la /shared-data/output/ || echo "Output directory is empty or inaccessible"
+          exit 0
```

Comment:
we cannot consider when there the file missing scenario as a success status. This is a complete failure where ProdSec team will lose a report for a package. Whether there are sast results or not, we must upload a file.